package/cloudshell/cm/customscript/domain/windows_script_executor.py

```diff
@@ -29,21 +29,34 @@ def __init__(self, logger, target_host, cancel_sampler):
         self.logger = logger
         self.cancel_sampler = cancel_sampler
         self.pool = ThreadPool(processes=1)
-        if target_host.connection_secured:
-            self.session = winrm.Session(target_host.ip, auth=(target_host.username, target_host.password), transport='ssl')
-        else:
+
+        # if parameter does not specify winrm_transport, try ssl, then fall back to http
+        if target_host.parameters.get('winrm_transport')=='ssl':
+            self.logger.info('SSL only WinRM session')
+            self.session = winrm.Session(target_host.ip, auth=(target_host.username, target_host.password), transport='ssl', server_cert_validation='ignore')
+        elif target_host.parameters.get('winrm_transport')=='http':
+            self.logger.info('http only WinRM session')
             self.session = winrm.Session(target_host.ip, auth=(target_host.username, target_host.password))
+        else:
+            self.logger.info('identifying whether host is ssl or http')
+            self.session = winrm.Session(target_host.ip, auth=(target_host.username, target_host.password), transport='ssl', server_cert_validation='ignore')
+            try:
+                self.session.run_cmd('@echo connected')
+            except ConnectionError:
+                self.session = winrm.Session(target_host.ip, auth=(target_host.username, target_host.password))
 
     def connect(self):
         try:
             uid = str(uuid4())
             result = self.session.run_cmd('@echo '+uid)
-            assert uid in result.std_out
+            stdout = result.std_out.decode('utf-8')
+            self.logger.info(stdout)
```

Comment:
personally, I dont like info dumps in log. some context is very helpful. but that code is working so your call i guess.